src/label-helpers.js

```diff
@@ -0,0 +1,55 @@
+function getLabelContent(label) {
+  let labelContent = label.getAttribute('value') || label.textContent
+  Array.from(label.querySelectorAll('textarea')).forEach(textarea => {
+    labelContent = labelContent.replace(textarea.value, '')
+  })
+  Array.from(label.querySelectorAll('select')).forEach(select => {
+    labelContent = labelContent.replace(select.textContent, '')
+  })
+  return labelContent
+}
+
+// Based on https://github.com/eps1lon/dom-accessibility-api/pull/352
+function getRealLabels(element) {
+  if (element.labels !== undefined) return element.labels
+
+  if (!isLabelable(element)) return []
+
+  const labels = element.ownerDocument.querySelectorAll('label')
+  return Array.from(labels).filter(label => label.control === element)
+}
+
+function isLabelable(element) {
+  return (
+    element.tagName.match(/BUTTON|METER|OUTPUT|PROGRESS|SELECT|TEXTAREA/) ||
+    (element.tagName === 'INPUT' && element.getAttribute('type') !== 'hidden')
+  )
+}
+
+function getLabels(container, element, {selector = '*'} = {}) {
+  const labelsId = element.getAttribute('aria-labelledby')
+    ? element.getAttribute('aria-labelledby').split(' ')
+    : []
+  let labelsList = labelsId.length
+    ? labelsId.map(labelId => {
+        const labellingElement = container.querySelector(`[id="${labelId}"]`)
+        return labellingElement
+          ? {content: getLabelContent(labellingElement), formControl: null}
+          : {content: '', formControl: null}
+      })
+    : Array.from(getRealLabels(element)).map(label => {
+        const textToMatch = getLabelContent(label)
+        const formControlSelector =
+          'button, input, meter, output, progress, select, textarea'
+        const labelledFormControl = Array.from(
+          label.querySelectorAll(formControlSelector),
+        ).filter(formControlElement => formControlElement.matches(selector))[0]
+
+        return {content: textToMatch, formControl: labelledFormControl}
+      })
+  labelsList = labelsList.filter(Boolean)
```

Comment:
I am not sure this line is meaningful written in this way. No element of `labelsList` could be `undefined` or `null`.
In the code you are replacing, this filter was performed to remove the empty element of `labelsList`, but I think you are doing this check at line 79 in `src/queries/label-text`.
So, maybe you could remove this line and keep the filter in `src/queries/label-text` or replace this line with `labelsList = labelsList.filter(label => Boolean(label.content))` and remove the filter in `src/queries/label-text`.

Reply:
Yes right I can remove